File: Algoritmos de Ordenamiento/Ordenamiento de Seleccion/SelectionSort.cpp
// Ordenamiento por selección
// Autor: Rivera Cota Jesús Crispín
// FACULTAD DE INGENIERÍA LOS MOCHIS
// Ing. de Software 201

#include <iostream>
using namespace std;

void selectionsort(int arr[], int n){
    for (int i = 0; i < n - 1; i++){
        int small = i;
        for (int j = i + 1; j < n; j++){
            if (arr[j] < arr[small]){
                small = j;
            }
        }
        if (small != i){
            int temp = arr[i];
            arr[i] = arr[small];
            arr[small] = temp;
        }
    }
}

int main(){
    int arr[] = {65, 26, 13, 23, 12};
    int n = sizeof(arr)/sizeof(arr[0]);
    cout << "El aray antes de ordenar es: " << endl;
    for (int i = 0; i < n; i++){
        cout << arr[i] << " ";
    }
    selectionsort(arr,n);
    cout << "\nEl array despuúes de ordenar es: " << endl;
    for (int i = 0; i < n; i++){
        cout << arr[i] << " ";
    }
}

File: Algoritmos de Ordenamiento/Ordenamiento de Seleccion/SelectionSort.py
# Ordenamiento por selección
# Autor: Rivera Cota Jesús Crispín
# FACULTAD DE INGENIERÍA LOS MOCHIS
# Ing. de Software 201

def selectionSort(a): #funcion para implementar el algoritmo de selección
    for i in range(len(a)):#recorre todo el arreglo
        small = i #indice del elemento mas pequeño
        for j in range(i+1, len(a)): #encuentra el elemento mas pequeño en el arreglo
            if a[small] > a[j]:#compara el elemento mas pequeño con el siguiente elemento
                small = j #actualiza el indice del elemento mas pequeño
                #intercambia el elemento mas pequeño con el primer elemento del arreglo
                a[i], a[small] = a[small], a[i] #intercambio de elementos
def printArray(a): #funcion para imprimir el arreglo
    for i in range(len(a)):#recorre todo el arreglo
        print(a[i], end=" ")
a=[65, 26, 13, 23, 12]
print("El arreglo antes de ser ordenado es: ")
printArray(a)
selectionSort(a)
print("\nEl arreglo después de ser ordenado es: ")
selectionSort(a)
printArray(a)